frontend/src/app/libs/common-components/components/input/input.component.ts

```diff
@@ -0,0 +1,283 @@
+
+	public inlineSyncValidators: ValidatorFn[] = null;
+	public inlineAsyncValidators: AsyncValidatorFn[] = null;
+
+	public errorsArray: string[] = [];
+
+	private syncValidators: ValidatorFn[] | null = null;
+	private asyncValidators: AsyncValidatorFn[] | null = null;
+
+	private destroy$: Subject<any> = new Subject();
+
+	private errorOnInlineValidators: string = 'All inline validators are ignored';
+
+	public get icon(): string {
+		return Boolean(this.secured) ? PasswordVisibility.On : PasswordVisibility.Off;
+	}
+
+	public get isDarkTheme(): boolean {
+		return this.theme === ComponentTheme.Dark;
+	}
+
+	public get isText(): boolean {
+		return this.type === InputType.Text ? true : false;
+	}
+
+	public get isEmail(): boolean {
+		return this.type === InputType.Email ? true : false;
+	}
+
+	public get isPassword(): boolean {
+		return this.type === InputType.Password ? true : false;
+	}
+
+	public get inputType(): string {
+		if (this.type !== InputType.Password) {
+			return this.type;
+		} else if (this.type === InputType.Password && this.secured) {
+			return InputType.Password;
+		} else if (this.type === InputType.Password && !this.secured) {
+			return InputType.Text;
+		}
+	}
+
+	public constructor(
+		@Optional() @Host() @SkipSelf() protected parentFormContainer: ControlContainer,
+		public elementRef: ElementRef,
+		public changeDetector: ChangeDetectorRef
+	) {}
+
+	public ngOnInit(): void {
+
+		this.formControlName = this.elementRef.nativeElement.getAttribute('formControlName');
+		if (this.formControlName != null && this.parentFormContainer != null) {
+
+			this.formControl = (this.parentFormContainer.control as FormGroup).controls[this.formControlName] as FormControl;
+			if (this.formControl === undefined) {
+				throw new Error(`Form control ${this.formControlName} is not registered in form group`);
+			}
+
+		} else {
+			this.formControl = new FormControl('');
+		}
+
+		this.subscribeToChanges();
+		this.checkInlineValidators();
+		this.setSyncValidator();
+		this.setAsyncValidator();
+		this.switchToDisable();
+	}
+	public ngAfterViewInit(): void {
+		setTimeout(() => {
+			if (Boolean(this.value)) {
+				this.formControl.setValue(this.value);
+			}
+		}, 0);
+	}
+
+	public ngOnDestroy(): void {
+		this.destroy$.next();
+		this.destroy$.complete();
+	}
+
+	public switchToDisable(): void {
+		if (Boolean(this.formControl) && this.disabled) {
+			this.formControl.disable();
+		}
+	}
+
+	public onTouched: any = () => undefined;
+
+	public onChange: any = () => undefined;
+
+	public writeValue(value: any): void {
+		if (!Boolean(this.value) ) {
+			this.value = value;
+		}
+		this.onChange(value);
+		if (this.formControl && this.formControl.value) {
+			this.onTouched();
+		}
+		this.valueChange.emit(value);
+		this.validate();
+	}
+
+	public onValueChange(value: any): void {
+		this.onTouched();
+		if (Boolean(this.value) && this.value !== value) {
+			this.value = value;
+			this.onChange(value);
```

Comment:
this line duplicates frequently, please move to separate method and replace all occurrences by using method